src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,137 @@
+package duke;
+
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+import duke.task.Deadline;
+
+import java.time.LocalDate;
+import java.util.Scanner;
+import java.io.IOException;
+import java.io.File;
+import java.io.FileWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Storage class to save information to hard disk.
+ */
+public class Storage {
+    private final String DEFAULT_PATH = "./duke.txt";
+    private final Path filePath;
+
+    public Storage() {
+        this.filePath = Paths.get(DEFAULT_PATH);
+        File file = new File(DEFAULT_PATH);
+        if (Files.notExists(this.filePath)) {
+            try {
+                System.out.println("created new file");
+                file.createNewFile();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    /**
+     * Constructor that takes in filepath to data file
+     * @param filePathStr
+     */
+    public Storage(String filePathStr) {
+        this.filePath = Paths.get(filePathStr);
+        File file = new File(DEFAULT_PATH);
+        if (Files.notExists(this.filePath)) {
+            try {
+                System.out.println("created new file");
+                file.createNewFile();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    /**
+     * Method to save TaskList to hard disk.
+     *
+     * @param taskList
```

Comment:
Perhaps consider standardizing on having or omitting "\n" between JavaDoc description and @param etc.

Reply:
Good Point! Thanks for the tip :)